### DISCUSSION TITLE
How to get Renovate to pickup container image updates that don't follow semver?

### DISCUSSION THREAD

**Discussion by javydekoning** [renovatebot/renovate#21626] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Renovate to auto pickup new container image

### What you tried so far.

Searched issues and docs, but got lost.

### Relevant debug logs

I use a few container images that do not follow semver such as: 

[ghcr.io/linuxserver/duplicati:development-v2.0.6.105-2.0.6.105_canary_2023-04-09-ls116](https://github.com/orgs/linuxserver/packages/container/duplicati/85640332?tag=development-v2.0.6.105-2.0.6.105_canary_2023-04-09-ls116)

[ghcr.io/onedr0p/plex:1.32.0.6950-8521b7d99](https://github.com/onedr0p/containers/pkgs/container/plex)

How should I change my Renovate config to successfully pick up new versions? Which part of the docs should I follow?

**Comment by rarkins** [comment]
Do you wish to stick with `development-` prefixes?

**Comment by javydekoning** [comment]
For the duplicati container, yes :)

**Comment by rarkins** [comment]
For `plex` you could probably fix this immediately by setting `versioning=loose`. For `duplicati` you might want to try `regex` versioning where you focus only on the initial `2.0.6` part as major, minor, patch

**Reply by javydekoning** [reply]
Thanks for pointing me in the right direction. 

Loose would help me sorting correctly. For Duplicati extracting `2.0.6.105-2.0.6.105_canary_2023-04-09-ls116` should then also work.

I was hoping a combination of those would work for Duplicati. I tried this:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "helm-values": {
    "fileMatch": ["(^|/)values\\.yaml$"],
    "versioning": "loose",
    "packageRules": [
      {
        "matchPackageNames": ["ghcr.io/linuxserver/duplicati"],
        "extractVersion": "^development\\-v(?<version>.*)$",
        "versioning": "loose"
      }
    ]
  },
  "kubernetes": {
    "fileMatch": ["k8s/.+/custom\\.yaml$"],
    "versioning": "loose"
  }
}
```

I also tried this approach, same result:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "helm-values": {
    "fileMatch": ["(^|/)values\\.yaml$"]
  },
  "kubernetes": {
    "fileMatch": ["k8s/.+/custom\\.yaml$"]
  },
  "packageRules": [
    {
      "matchManagers": ["helm-values"],
      "matchPackageNames": ["ghcr.io/linuxserver/duplicati"],
      "extractVersion": "^development\\-(?<version>.+?)$",
      "versioning": "loose"
    }, {
      "matchManagers": ["helm-values"],
      "matchPackageNames": ["ghcr.io/onedr0p/plex"],
      "versioning": "loose"
    }
  ]
}

```

But it fails to extract the version.

```
Dependency ghcr.io/linuxserver/duplicati has unsupported/unversioned value development-v2.0.6.105-2.0.6.105_canary_2023-04-09-ls116 (versioning=loose)

    "helm-values": [
      {
        "deps": [
          {
            "depName": "ghcr.io/linuxserver/duplicati",
            "currentValue": "development-v2.0.6.105-2.0.6.105_canary_2023-04-09-ls116",
            "datasource": "docker",
            "replaceString": "development-v2.0.6.105-2.0.6.105_canary_2023-04-09-ls116",
            "versioning": "loose",
            "autoReplaceStringTemplate": "{{newValue}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "updates": [],
            "packageName": "ghcr.io/linuxserver/duplicati",
            "warnings": [],
            "skipReason": "invalid-value"
          }
        ],
        "packageFile": "k8s/cluster-apps/duplicati/values.yaml"
      },
```

**Reply by javydekoning** [reply]
As I wasn't able to get this to work, I used digest pinning instead.

If anyone ever stumbles on this, this is the tag I used:

```
Tag: "2.0.6-development@sha256:3641761593478e8d57f16755bf74ba32994bcccf3d3ad81f5f257d98e575f3e8"
```
